Update cors.mdx (#4311)
apollographql/router: docs/source/configuration/cors.mdx

    * Use this option if your supergraph is a public API with arbitrarily many web app consumers.
    * With this option enabled, the router sends the [wildcard (`*`)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#directives) value for the `Access-Control-Allow-Origin` header. This enables _any_ website to initiate browser connections to it (but they can't provide cookies or other credentials).

> * You _must_ use either `origins`, `match_origins`, or the combination of both options if clients need to [authenticate their requests with cookies](#passing-credentials). Note that `origins` will be evaluated before `match_origins`.

Reviewer: Does the `#passing-credentials` section need to be updated to reflect using `origins` and/or `match_origins`?

```suggestion
* If clients need to [authenticate their requests with cookies](#passing-credentials), you _must_ use either `origins`, `match_origins`, or the combination of both options. When using both options, note that `origins` is evaluated before `match_origins`.
```

> The following snippet includes an example of each option (use either `allow_any_origin`, or `origins + match_origins`):

Reviewer: Do the docs explain elsewhere about `use either allow_any_origin, or origins + match_origins`? I can't find another section, and it seems it's worthwhile to do so, rather than just in this parenthetical.

```suggestion
The following snippet includes an example of each option (use either `allow_any_origin`, or `origins` and/or `match_origins`):
```

Author: I did not see explanation elsewhere, but agree it would be worthwhile.
